src/Components/Header/Header.tsx

```diff
@@ -0,0 +1,132 @@
+import './Header.scss';
+import { NavLink, useLocation } from 'react-router-dom';
+import { HeaderSearch } from './HeaderSearch';
+import HeartImage from './HeaderImage/heart.svg';
+import BasketImage from './HeaderImage/basket.svg';
+import LogoImage from './HeaderImage/LOGO.svg';
+import { useFavoriteContext } from '../../FavoriteContext';
+
+interface HeaderProps {
+  searchValue: string;
+  setSearchValue: (search: string) => void;
+}
+
+export const Header = ({ searchValue, setSearchValue }: HeaderProps) => {
+  const location = useLocation();
+  const isHomePage = location.pathname === '/';
+  const contacts = location.pathname === '/contacts';
+  const rights = location.pathname === '/rights';
+  const phone = location.pathname === '/phones';
+  const tablets = location.pathname === '/tablets';
+  const accessories = location.pathname === '/accessories';
+  const basket = location.pathname === '/cart';
+  const favorite = location.pathname === '/favorites';
+  // const [showSearch, setShowSearch] = useState(!isHomePage);
+
+  let search = '';
+
+  if (!isHomePage && !contacts && !rights && !favorite && !basket) {
```

Comment:
```suggestion
 
  // const [showSearch, setShowSearch] = useState(!isHomePage);

  const ARRAY = ['/', '/rights', ....]

  let search = '';

  if (!ARRAY.includes(location.pathname)) {
```
you can rename this array and move out of the ccomponent